gramps/gen/lib/grampstype.py

```diff
@@ -165,41 +179,27 @@ def __set_tuple(self, value):
         self.__value = val
         self.__string = strg
 
-    def __set_int(self, value):
+    @set.register
+    def set_int(self, value: int):
         "Set the value/string properties from an integer."
         self.__value = value
         self.__string = ""
 
-    def __set_instance(self, value):
-        "Set the value/string properties from another grampstype."
-        self.__value = value.value
-        if self.__value == self._CUSTOM:
-            self.__string = value.string
-        else:
-            self.__string = ""
-
-    def __set_str(self, value):
+    @set.register
+    def set_str(self, value: str):
         "Set the value/string properties from a string."
         self.__value = self._S2IMAP.get(value, self._CUSTOM)
         if self.__value == self._CUSTOM:
             self.__string = value
         else:
             self.__string = ""
 
-    def set(self, value):
-        "Set the value/string properties from the passed in value."
-        if isinstance(value, dict):
-            self.__set_dict(value)
-        elif isinstance(value, tuple):
-            self.__set_tuple(value)
-        elif isinstance(value, int):
-            self.__set_int(value)
-        elif isinstance(value, self.__class__):
-            self.__set_instance(value)
-        elif isinstance(value, str):
-            self.__set_str(value)
+    def set_instance(self, value):
```

Comment:
This should be private.

Reply:
See discussion below on [`def set_instance(self, value: GrampsType):`](https://github.com/gramps-project/gramps/pull/1830#discussion_r1912466188)

Reply:
It still doesn't need to be public; it's only used by something that is added on to the class. Note however, if you use double underscores, you will need to use the mangled name: `_GrampsType__set_instance`.

Reply:
This is addressed in the most recent commit